### PR TITLE
Add suppport for AWS IAM roles

### PR DESCRIPTION
**Pull request by hoylu-royberg**
Added support for using AWS instance profile credentials, with or without specifying an IAM role name.

### CODE REVIEW

**Comment by hoylu-royberg** [comment]
@KSemenenko I have created a draft PR for https://github.com/managedcode/Storage/issues/96. It's a draft because I wanted to hear your opinion about the approach used. If you find it acceptable, I will try to add some tests. I saw that there were no integration tests for AWS. I've used https://github.com/adobe/S3Mock before as an emulator for S3, but I see that it doesn't support IAM roles unfortunately, so might have to look at using https://github.com/localstack/localstack instead.

An alternative approach could be to add a nullable `AWSCredentials` property in `AWSStorageOptions`, and let library consumer set it to the credentials provider they want to use, and where the default is to use `BasicAWSCredentials`.

File: ManagedCode.Storage.Aws/Options/AWSStorageOptions.cs [hunk 1]
```diff
@@ -3,12 +3,49 @@
 
 namespace ManagedCode.Storage.Aws.Options;
 
+/// <summary>
+///     Configuration options for AWS S3 storage.
```
Comment (hoylu-royberg) [inline ManagedCode.Storage.Aws/Options/AWSStorageOptions.cs:7]: Personally I don't use indentation for the XML comments, but I saw that this was used other places in this repository.

**Comment by KSemenenko** [comment]
Is amazing PR!

File: ManagedCode.Storage.Aws/Options/AWSStorageOptions.cs (hunk 1 above)
Comment (KSemenenko) [reply ManagedCode.Storage.Aws/Options/AWSStorageOptions.cs:7]: I do this in copilot, so no worrirs :)

**Comment by KSemenenko** [comment]
we use lockalstack. Main is not 
latest branch, let me merge some stuff, and we will have striped integrations tests. 

This this amazing job!

**Comment by KSemenenko** [comment]
I updated main, can you please updte your PR? now tests are availalbe

**Comment by hoylu-royberg** [comment]
@KSemenenko Thank you for your kind words, and for updating main! ☺️ I have pulled your latest changes, and I see that localstack is there, so this is awesome. I will work on some tests tomorrow (Monday).

**Comment by hoylu-royberg** [comment]
@KSemenenko Are you OK with pulling in the NuGet package `AWSSDK.IdentityManagement` for the `ManagedCode.Storage.Tests` project? I need it in order to create an IAM role and a policy, and then bind that policy to an instance profile, from code. I was hoping to instead use [LocalStack's IAM Policy Stream](https://docs.localstack.cloud/user-guide/security-testing/iam-policy-stream/), but so far I have not found a way to do that from code.

**Comment by KSemenenko** [comment]
@hoylu-royberg of course ass any packages you need

**Comment by hoylu-royberg** [comment]
@KSemenenko I have not been able to write integration tests for proving that the IAM role code changes are working, because it turns out that IAM enforcement is a LocalStack Pro feature, i.e. requires a LocalStack license. See these links:

- https://docs.localstack.cloud/user-guide/security-testing/iam-enforcement/ (tagged with "Pro Image" at the top)
- https://github.com/localstack/localstack/issues/7183#issuecomment-1315656000 (LocalStack developer says _"IAM Enforcement is a LocalStack Pro feature, so it will not work without a valid license"_)

So although I wrote the code required to:

- Create IAM role
- Create IAM user
- Create an S3 policy
- Attach policy to role
- Attach policy to user
- Create instance profile

It's all rather pointless when a test will pass regardless if a valid role is used or not. The test has then not proved anything 😞

I have however tested my changes in a real AWS environment, and it worked -- both with explicit and implicit instance profile. So I'm marking this PR as ready for review now. Let me know if you want me to add the `AwsHelper.cs` class I wrote, that performs the steps mentioned above. I didn't include it, since it's not used for anything now.

**Comment by KSemenenko** [comment]
Thanks, I trust you :)

**Pull request merged**